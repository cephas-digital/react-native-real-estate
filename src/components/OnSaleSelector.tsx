import React, { Component } from "react";
//@ts-ignore
import { RadioButton } from "react-native-btr";
import { View, Text } from "react-native";
import { ListingFilters } from "../interfaces";

interface MyProps {
  setFilters(filters: ListingFilters): void;
  filters: ListingFilters;
}

export default class OnSaleSelector extends Component<MyProps> {
  onButtonPress(onSaleValue: number) {
    this.props.setFilters({ onSale: onSaleValue });
  }

  generateButtons(): Array<Element> {
    const buttons: Array<Element> = [];
    [
      {
        text: "All",
        value: 2,
        checked: (props: any) =>
          props.filters.onSale === 2 ||
          props.filters.onSale === null ||
          typeof props.filters.onSale === "undefined"
      },
      {
        text: "Yes",
        value: 1,
        checked: (props: any) => props.filters.onSale === 1
      },
      {
        text: "No",
        value: 0,
        checked: (props: any) => props.filters.onSale === 0
      }
    ].forEach((button, index) => {
      buttons.push(
        <RadioButton
          key={index}
          checked={button.checked(this.props)}
          color={"#484"}
          onPress={() => this.onButtonPress(button.value)}
          size={10}
        >
          <Text style={[{ margin: 10, color: "#484" }]}>{button.text}</Text>
        </RadioButton>
      );
    });

    return buttons;
  }

  render() {
    return (
      <View
        style={{
          alignItems: "center",
          marginVertical: 10
        }}
      >
        <Text
          style={{
            marginBottom: 10,
            fontSize: 20,
            justifyContent: "center"
          }}
        >
          Is on Sale?
        </Text>
        <View>{this.generateButtons()}</View>
      </View>
    );
  }
}
