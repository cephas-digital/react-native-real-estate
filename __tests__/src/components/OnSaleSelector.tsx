import React from "react";
import renderer from "react-test-renderer";
import OnSaleSelector from "../../../src/components/OnSaleSelector";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <OnSaleSelector
        filters={{}}
        setFilters={filters => console.log(filters)}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
