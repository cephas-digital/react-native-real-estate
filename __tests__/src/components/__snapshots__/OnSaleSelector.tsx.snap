// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "justifyContent": "center",
        "marginBottom": 10,
      }
    }
  >
    Is on Sale?
  </Text>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#484",
            "borderRadius": 20,
            "borderWidth": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#484",
              "borderRadius": 10,
              "margin": 2,
              "padding": 10,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#484",
              "margin": 10,
            },
          ]
        }
      >
        All
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#484",
            "borderRadius": 20,
            "borderWidth": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#0000",
              "borderRadius": 10,
              "margin": 2,
              "padding": 10,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#484",
              "margin": 10,
            },
          ]
        }
      >
        Yes
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderColor": "#484",
            "borderRadius": 20,
            "borderWidth": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#0000",
              "borderRadius": 10,
              "margin": 2,
              "padding": 10,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#484",
              "margin": 10,
            },
          ]
        }
      >
        No
      </Text>
    </View>
  </View>
</View>
`;
